homework4/task4.py

```diff
@@ -0,0 +1,9 @@
+import re
+with open('index.html', 'r') as f:
+    html_file = f.read()
+change_code = re.findall(r'<div\s+id="(?P<id>[^"]+)">\s*<a\s+href="(?P<href>[^"]+)">\s*(?P<text>[^<]+)\s*</a>',
+                         html_file, flags=0)
```

Comment:
let it be. But this regex is too complicated. Better to slit it to several more simple regexes